File: python_comparison/file.py
import time

t0 = time.time()
with open("input.txt", "r") as f:
    out = []
    for line in f.read().strip().split("\n"):
        a, b, c = [float(x) for x in line.split(",")]
        n = a * b + c
        if len(out) and out[-1] > n:
            out.append(n - out[-1])
        out.append(n)
    print(time.time() - t0, "seconds")

File: python_comparison/file.ts
import fs from 'node:fs'

const t0 = performance.now()
fs.readFile('input.txt', 'utf8', function (err, data) {
  if (err) throw err
  const out = []
  for (const line of data.trim().split('\n')) {
    const [a, b, c] = line.split(',').map(Number)
    const n = a * b + c
    if (out.length && out[out.length - 1] > n) {
      out.push(n - out[out.length - 1])
    }
    out.push(n)
  }
  console.log((performance.now() - t0) / 1e3, 'seconds')
})
